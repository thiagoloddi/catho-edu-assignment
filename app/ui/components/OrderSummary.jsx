import React from 'react';
import injectSheet from 'react-jss';
import propTypes from 'prop-types';

import { formatCurrency } from '../../utils/numbers';

const OrderSummary = ({ classes, loading, sum }) => {

  const renderSum = () => {
    if(loading) {
      return <img className={classes.sumLoading} src="/assets/progress.gif" />
    } else {
      return formatCurrency(sum / 100);
    }
  }

  return (
    <div className={classes.container}>
      <div className={classes.priceField}>
        <div className={classes.heading}>FINAL PRICE:</div>
        <div className={classes.price}>{renderSum()}</div>
      </div>
    </div>
  )
}

OrderSummary.propTypes = {
	loading: propTypes.bool,
	sum: propTypes.number.isRequired
};

const style = {
  container: {
    position: 'relative',
    display: 'inline-block',
    width: '100%',
    border: '2px solid rgba(0, 0, 0, .1)',
    padding: '10px 20px',
    marginTop: '20px',
  },
  heading: {
    fontWeight: 'bold'
  },
  priceField: {
    width: '100%',
    '& > *': {
      display: 'inline-block',
      fontSize: '20px',
    }
  },
  price: {
    float: 'right'
  },
  sumLoading: {
    float: 'right',
    height: '20px',
    width: '20px'
  }
}

export default injectSheet(style)(OrderSummary);