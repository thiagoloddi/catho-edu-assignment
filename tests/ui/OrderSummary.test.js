import React from 'react';
import renderer from 'react-test-renderer';
import OrderSummary from '../../app/ui/components/OrderSummary';

it('renders correctly', () => {
  const tree = renderer
    .create(<OrderSummary sum={10}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});