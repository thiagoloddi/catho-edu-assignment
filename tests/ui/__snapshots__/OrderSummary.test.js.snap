// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="OrderSummary-container-0-1-1"
>
  <div
    className="OrderSummary-priceField-0-1-3"
  >
    <div
      className="OrderSummary-heading-0-1-2"
    >
      FINAL PRICE:
    </div>
    <div
      className="OrderSummary-price-0-1-4"
    >
      $0.10
    </div>
  </div>
</div>
`;
